### PR TITLE
[Rec-IM] Error Handling

### PR DESCRIPTION
**Pull request by silasw2**
This PR adds Snackbars to all form completions, so that they will no longer hang on cancel, and give helpful error messages to the user.

### CODE REVIEW

File: src/views/RecIM/views/Match/index.jsx [hunk 1]
```diff
@@ -356,20 +357,23 @@ const Match = () => {
...
+
             {match.Scores.length !== 0 && (
               <EditMatchStatsForm
                 match={match}
                 setMatch={setMatch}
-                closeWithSnackbar={(status) => {
-                  handleFormSubmit(status, setOpenEditMatchStatsForm);
+                onClose={() => {
+                  setAnchorEl(null);
                 }}
```
Comment (amos-cha) [inline src/views/RecIM/views/Match/index.jsx:375]:
```suggestion
                onClose={() => setAnchorEl(null)}
```

File: src/views/RecIM/views/Match/index.jsx [hunk 2]
```diff
@@ -356,20 +357,23 @@ const Match = () => {
               </MenuItem>
             </Menu>
             <MatchForm
-              closeWithSnackbar={(status) => {
-                handleFormSubmit(status, setOpenMatchInformationForm);
+              createSnackbar={createSnackbar}
+              onClose={() => {
+                setReload((prev) => !prev);
               }}
```
Comment (amos-cha) [inline src/views/RecIM/views/Match/index.jsx:363]:
```suggestion
              onClose={() => setReload((prev) => !prev)}
```

File: src/views/RecIM/views/Admin/index.jsx [hunk 3]
```diff
@@ -172,7 +169,10 @@ const Admin = () => {
       </Card>
       <SurfaceForm
         surface={surface}
-        closeWithSnackbar={handleClose}
+        createSnackbar={createSnackbar}
+        onClose={async () => {
+          setSurfaces(await getSurfaces());
+        }}
```
Comment (amos-cha) [inline src/views/RecIM/views/Admin/index.jsx:175]:
```suggestion
        onClose={async () =>  setSurfaces(await getSurfaces())}
```

File: src/views/RecIM/views/Activity/index.jsx [hunk 4]
```diff
@@ -359,69 +359,54 @@ const Activity = () => {
...
-                setReload(!reload);
-                setSnackbar({
-                  message: 'Added participant successfully',
-                  severity: 'success',
-                  open: true,
-                });
+              createSnackbar={createSnackbar}
+              onClose={() => {
+                setReload((prev) => !prev);
               }}
```
Comment (amos-cha) [inline src/views/RecIM/views/Activity/index.jsx:404]:
```suggestion
              onClose={() => setReload((prev) => !prev)}
```

File: src/views/RecIM/views/Activity/index.jsx [hunk 5]
```diff
@@ -359,69 +359,54 @@ const Activity = () => {
...
-                setOpenImageOptions={setOpenImageOptions}
-              />
-            )}
-
+            <ImageOptions
+              category={'Activity'}
+              createSnackbar={createSnackbar}
+              onClose={() => {
+                setReload((prev) => !prev);
+              }}
```
Comment (amos-cha) [inline src/views/RecIM/views/Activity/index.jsx:395]:
```suggestion
              onClose={() => setReload((prev) => !prev)}
```

File: src/views/RecIM/views/Activity/index.jsx [hunk 6]
```diff
@@ -359,69 +359,54 @@ const Activity = () => {
...
               setOpenSeriesForm={(bool) => setOpenCreateSeriesForm(bool)}
               activityID={activity.ID}
               existingActivitySeries={activity.Series}
             />
             <TeamForm
-              closeWithSnackbar={(teamID, status) => {
-                handleFormSubmit(status, setOpenTeamForm);
+              onClose={(teamID) => {
                 navigate(`team/${teamID}`);
               }}
```
Comment (amos-cha) [inline src/views/RecIM/views/Activity/index.jsx:384]:
```suggestion
              onClose={(teamID) => navigate(`team/${teamID}`)}
```

File: src/views/RecIM/views/Activity/index.jsx [hunk 7]
```diff
@@ -359,69 +359,54 @@ const Activity = () => {
...
 
-            {/* forms and dialogs */}
-
             <ActivityForm
               activity={activity}
-              closeWithSnackbar={(status) => {
-                handleFormSubmit(status, setOpenActivityForm);
+              onClose={() => {
+                setReload((prev) => !prev);
               }}
```
Comment (amos-cha) [inline src/views/RecIM/views/Activity/index.jsx:366]:
```suggestion
              onClose={() => setReload((prev) => !prev)}
```

File: src/views/RecIM/views/Activity/components/ScheduleList/index.jsx [hunk 8]
```diff
@@ -343,38 +352,35 @@ const ScheduleList = ({ isAdmin, series, activityID, reload, setReload, activity
...
+        }}
+        openSeriesScheduleForm={openSeriesScheduleForm}
+        setOpenSeriesScheduleForm={(bool) => setOpenSeriesScheduleForm(bool)}
+        seriesID={series.ID}
+      />
+      <MatchForm
+        createSnackbar={createSnackbar}
+        onClose={() => {
+          setReload((prev) => !prev);
+        }}
```
Comment (amos-cha) [inline src/views/RecIM/views/Activity/components/ScheduleList/index.jsx:379]:
```suggestion
      <SeriesForm
        createSnackbar={createSnackbar}
        onClose={() => setReload((prev) => !prev)}
        openSeriesForm={openEditSeriesForm}
        setOpenSeriesForm={(bool) => setOpenEditSeriesForm(bool)}
        activityID={series.ActivityID}
        series={series}
        activityTeams={activityTeams}
      />
      <SeriesScheduleForm
        createSnackbar={createSnackbar}
        onClose={() => setReload((prev) => !prev)}
        openSeriesScheduleForm={openSeriesScheduleForm}
        setOpenSeriesScheduleForm={(bool) => setOpenSeriesScheduleForm(bool)}
        seriesID={series.ID}
      />
      <MatchForm
        createSnackbar={createSnackbar}
        onClose={() => setReload((prev) => !prev)}
```

File: src/views/RecIM/views/Team/index.jsx [hunk 9]
```diff
@@ -295,8 +317,9 @@ const Team = () => {
               <Typography variant="body1">This action cannot be undone.</Typography>
             </GordonDialogBox>
             <InviteParticipantForm
-              closeWithSnackbar={(status) => {
-                handleInviteParticipantForm(status);
+              createSnackbar={createSnackbar}
+              onClose={() => {
+                setReload((prev) => !prev);
               }}
```
Comment (amos-cha) [inline src/views/RecIM/views/Team/index.jsx:323]:
```suggestion
              onClose={() => setReload((prev) => !prev)}
```

File: src/views/RecIM/views/Team/index.jsx [hunk 10]
```diff
@@ -269,12 +267,36 @@ const Team = () => {
             <ImageOptions
               category={'Team'}
               component={team}
-              closeWithSnackbar={(status) => {
-                handleOpenImageOptionsSubmit(status, setOpenImageOptions);
+              createSnackbar={createSnackbar}
+              onClose={() => {
+                setReload((prev) => !prev);
               }}
```
Comment (amos-cha) [inline src/views/RecIM/views/Team/index.jsx:273]:
```suggestion
              onClose={() => setReload((prev) => !prev)}
```

File: src/views/RecIM/views/Team/index.jsx [hunk 11]
```diff
@@ -257,8 +254,9 @@ const Team = () => {
 
             {/* forms and dialogs */}
             <TeamForm
-              closeWithSnackbar={(status) => {
-                handleTeamForm(status);
+              createSnackbar={createSnackbar}
+              onClose={() => {
+                setReload((prev) => !prev);
               }}
```
Comment (amos-cha) [inline src/views/RecIM/views/Team/index.jsx:260]:
```suggestion
              onClose={() => setReload((prev) => !prev)}
```

**Review by amos-cha — CHANGES_REQUESTED** [review]
Amazing work but there is a consistency issue regarding your 
`onClose={() => setReload((prev) => !prev)}
I think I've tracked them all down, but you're welcome to look through to make sure I haven't missed any.

**Review by amos-cha — APPROVED** [review]
A batch commit would have sufficed but this is fine too 😆

**Comment by silasw2** [comment]
> A batch commit would have sufficed but this is fine too 😆

Upping the numbers

**Pull request merged**